### DISCUSSION TITLE
Add time(hours) to chosenDate from Rowy columns

### DISCUSSION THREAD

**Discussion by prasmaran** [AntlerVC/firetable#1093] [Support / Q&A] [answered]
Hi there community ~

I tried to use derivatives. I am trying to add hours <based on package chosen by user> to the bookingTime <dd/mm/yyyy hh:mm>. But I am getting an Invalid date error. Am I using the timestamp wrongly here? 

Below is the code snippet I am using. Output field is Long Text
![rowy_row](https://user-images.githubusercontent.com/78822062/214540444-ce5eb87f-5abe-4d2c-a6f0-22885cec013c.PNG)

```
  const moment = require('moment');
  const lockedTime = row.lockTime;

  var timeIncrements = { "1_HOUR": 1, "2_HOUR": 2, "3_HOUR": 3, "4_OPEN": 0 };
  var increment = timeIncrements[row.package];

  let newDate = moment(lockedTime, 'DD/MM/YYYY HH:mm');
  newDate.add(increment, 'hours');
  return String(newDate.format('DD/MM/YYYY HH:mm'));
```

Appreciate your insight here. Thanks a lot

**Comment by bofeiw** [comment, marked as answer]
It says invalid date because it is expecting a Date object, not string. You need to return a Javascript Date object rather than string.

How to transform moment object to Date: https://momentjs.com/docs/#/displaying/as-javascript-date/

**Reply by prasmaran** [reply]
Hi @bofeiw ~ appreciate your reply

But I did pass the Object here in the argument. Or is it due to the Timestamp format of Firestore that I am not converting the date correctly? Thank you

```
  const moment = require('moment');
  const lockedTime = row.lockTime; // **This one is the Object right?**
  let newDate = moment(lockedTime, 'DD/MM/YYYY HH:mm'); // **Passing the Object as the argument here**
  newDate.add(increment, 'hours');
  return String(newDate.format('DD/MM/YYYY HH:mm'));
```

**Reply by bofeiw** [reply]
If you set the column type to Date, you must return a Date object, not string. 
e.g.
```js
return newDate.toDate();
```

If you want to display the string in your preferred format, set the column type to short text instead.

**Reply by prasmaran** [reply]
Thanks for hint right there @bofeiw 

I was able to get the thing I wanted. Thanks a lot

```
try {
    const doc = await bookingRef.get();
    if (!doc.exists) {
      return "The document does not exist.";
    } else {
      let time = doc.data().lockTime
      const dateString = time.toDate();
      const date = moment(dateString);
      return `Ending time: ${date.format("DD/MM/YYYY HH:mm")}`;
    }
  } catch (error) {
    return `Error getting document ${error}`;
  }
```
